### PR TITLE
Review permissions for boxes

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

File: deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxListService.java [hunk 1]
```diff
@@ -213,7 +214,8 @@ public AttributedList<Path> list(final Path directory, final ListProgressListene
                     list.add(documents.withAttributes(attributes.find(documents)));
                 }
                 if(box.getBoxPolicy().isCanAccessTrash()) {
-                    final Path trash = new Path(directory, containerService.getPinnedLocalization(TRASH), EnumSet.of(Path.Type.directory, Path.Type.volume));
+                    final Path trash = new Path(directory, containerService.getPinnedLocalization(TRASH), EnumSet.of(Path.Type.directory, Path.Type.volume),
+                            new PathAttributes().withHidden(true));
```
Comment (chenkins) [inline deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxListService.java:218]: Add test coverage.

File: deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxPathContainerService.java [hunk 2]
```diff
@@ -49,7 +49,16 @@ public boolean isContainer(final Path file) {
         if(file.isRoot()) {
             return false;
         }
-        return file.isDirectory() && (file.getParent().isRoot() || file.getParent().getParent().isRoot() || file.getParent().getParent().isRoot() || file.getParent().getParent().getParent().isRoot());
+        if(this.isDeepbox(file)) {
```
Comment (chenkins) [inline deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxPathContainerService.java:52]:
```suggestion
        if(!file.isDirectory()) {
            return false;
        }
        if(this.isDeepbox(file)) {
```

**Review by chenkins — CHANGES_REQUESTED** [review]
Add test coverage.

File: deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java [hunk 3]
```diff
@@ -163,8 +163,12 @@ public void testListInbox() throws Exception {
...
     public void testListTrash() throws Exception {
         final DeepboxIdProvider nodeid = new DeepboxIdProvider(session);
-        final Path trash = new Path("/ORG 4 - DeepBox Desktop App/ORG3:Box1/Trash", EnumSet.of(Path.Type.directory));
-        new DeepboxListService(session, nodeid).list(trash, new DisabledListProgressListener()); // assert no fail
+        final Path box = new Path("/ORG 4 - DeepBox Desktop App/ORG3:Box1", EnumSet.of(Path.Type.directory, Path.Type.volume));
+        final AttributedList<Path> list = new DeepboxListService(session, nodeid).list(box, new DisabledListProgressListener());
+        final Path trash = new Path(box, "Trash", EnumSet.of(Path.Type.directory, Path.Type.volume));
+        assertNotNull(list.find(new SimplePathPredicate(trash)));
+        assertTrue(list.find(new SimplePathPredicate(trash)).attributes().isHidden());
+        assertNotSame(AttributedList.EMPTY, new DeepboxListService(session, nodeid).list(trash, new DisabledListProgressListener())); // assert no fail
```
Comment (chenkins) [inline deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java:171]:
```suggestion
        assertNotSame(AttributedList.EMPTY, new DeepboxListService(session, nodeid).list(trash, new DisabledListProgressListener()));
```

File: deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java [hunk 4]
```diff
@@ -372,6 +376,7 @@ public void testListChildrenTrash() throws Exception {
         final DeepboxIdProvider nodeid = new DeepboxIdProvider(session);
         final Path folder = new Path("/ORG 4 - DeepBox Desktop App/ORG3:Box1/Trash/", EnumSet.of(Path.Type.directory, Path.Type.volume));
         final PathAttributes attributes = new DeepboxAttributesFinderFeature(session, nodeid).find(folder);
+        assertTrue(attributes.isHidden());
         assertTrue(new BoxRestControllerApi(session.getClient()).getBox(ORG4, ORG4_BOX1).getBoxPolicy().isCanAddFilesRoot());
         assertTrue(attributes.getAcl().get(new Acl.CanonicalUser()).contains(CANLISTCHILDREN));
         // assert no fail
```
Comment (chenkins) [inline deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java:382]:
```suggestion
```

**Review by chenkins — APPROVED** [review]
Lgtm. Remove non-necessary code comments.

File: deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java (hunk 3 above)
Comment (dkocher) [reply deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java:171]: Comment still valid.

File: deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java (hunk 4 above)
Comment (dkocher) [reply deepbox/src/test/java/ch/cyberduck/core/deepbox/DeepboxListServiceTest.java:382]: Comment still valid.

**Review by ylangisc — APPROVED** [review]

**Pull request merged**